### ISSUE TITLE
Helper classes for unit testing serverless applications

### ISSUE THREAD

**Issue by hoegertn** [open-constructs/cdk-serverless#302]
- API GW Request/Event Builder
- Lambda Context object
- DataStore Mock
- expect helper for HTTP/REST responses

**Comment by mgochoa** [comment]
Hello. You need any help with it? I really liked this project.
These classes need to be in the tests folder and just mock all the things listed here right?

**Comment by hoegertn** [comment]
Yes they should go in the https://github.com/taimos/cdk-serverless/tree/main/src/tests folder and be available to users of the library.

Maybe create a class UnitTestUtil similar to the IntegUtil and have a factory method for API GW requests, a Lambda context object. Also a Mock object to be used with aws-cdk-mock for the DynamoDB would be great. And utils to expect the responses of API GW handlers. It would be good to have several independant PRs I think. Pick whatever you want and let me know so we do not duplicate the tasks.

On another note I would be happy to hear any feedback about the lib and where you are using it.

**Comment by hoegertn** [comment]
You might want to look into https://github.com/ashiina/lambda-local

We can use that or incorporate come ideas from there. I used it in the past a lot.
